[hue] "Cannot read field "ct" because "capabilities.control"" after upgrade to 4.1.0 from 3.x

After upgrading to openHAB 4.1.0 from 3.x (sorry can not say exact release) I have started getting the following error from my HUE-Bridge (deCONZ) in the openhab-log:

```
2024-01-02 08:44:32.973 [ERROR] [core.thing.internal.ThingManagerImpl] - Exception occurred during notification about bridge status change on thing 'hue:0010:huebridge:3': Cannot read field "ct" because "capabilities.control"
java.lang.NullPointerException: Cannot read field "ct" because "capabilities.control" is null
        at org.openhab.binding.hue.internal.handler.HueLightHandler.initializeCapabilities(HueLightHandler.java:197) ~[?:?]
        at org.openhab.binding.hue.internal.handler.HueLightHandler.initializeThing(HueLightHandler.java:148) ~[?:?]
        at org.openhab.binding.hue.internal.handler.HueLightHandler.bridgeStatusChanged(HueLightHandler.java:129) ~[?:?]
        at org.openhab.core.thing.internal.ThingManagerImpl.lambda$11(ThingManagerImpl.java:822) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
```

And the Things changes to "ERROR:BRIDGE" status while the HUE-Bridge "stays" in ONLINE status:

![image](https://github.com/openhab/openhab-addons/assets/36599761/baf821ff-8102-4982-b712-2d2d73ce429c)

![image](https://github.com/openhab/openhab-addons/assets/36599761/473ea357-5491-4426-b757-cedc9bdb67a5)


After a reboot everything is once again green for some time, then it fails again.


## My Environment
- OS: Fedora Linux 37 (Server Edition)
- openhab.noarch  4.1.0-1 @openHAB-Stable
- openhab-addons.noarch    4.1.0-1  @openHAB-Stable
- java-17-openjdk.x86_64     1:17.0.8.0.7-1.fc37     @updates
- java-17-openjdk-headless.x86_64     1:17.0.8.0.7-1.fc37    @updates
- nodejs.x86_64     1:18.18.2-1.fc37    @updates
- deCONZ - Phoscon App Gateway version 2.24.2, Firmware 26720700


I have deleted and "reinstalled" the HUE-Bridge-Binding and deCONZ-Bindig , but this did not help.

Any suggestions for a solution or "workaround"?

-- 
Dan

--- a/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueLightHandler.java
+++ b/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueLightHandler.java
@@ -26,6 +26,7 @@ import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.hue.internal.api.dto.clip1.Capabilities;
 import org.openhab.binding.hue.internal.api.dto.clip1.ColorTemperature;
+import org.openhab.binding.hue.internal.api.dto.clip1.Control;
 import org.openhab.binding.hue.internal.api.dto.clip1.FullLight;
 import org.openhab.binding.hue.internal.api.dto.clip1.State;
 import org.openhab.binding.hue.internal.api.dto.clip1.StateUpdate;
@@ -191,26 +192,34 @@ public class HueLightHandler extends BaseThingHandler implements HueLightActions
     }
 
     private void initializeCapabilities(@Nullable FullLight fullLight) {
-        if (!capabilitiesInitializedSuccessfully && fullLight != null) {
-            Capabilities capabilities = fullLight.capabilities;
-            if (capabilities != null) {
-                ColorTemperature ct = capabilities.control.ct;
-                if (ct != null) {
-                    colorTemperatureCapabilties = ct;
-
-                    // minimum and maximum are inverted due to mired/Kelvin conversion!
-                    StateDescriptionFragment stateDescriptionFragment = StateDescriptionFragmentBuilder.create()
-                            .withMinimum(new BigDecimal(LightStateConverter.miredToKelvin(ct.max))) //
-                            .withMaximum(new BigDecimal(LightStateConverter.miredToKelvin(ct.min))) //
-                            .withStep(new BigDecimal(100)) //
-                            .withPattern("%.0f K") //
-                            .build();
-                    stateDescriptionProvider.setStateDescriptionFragment(
-                            new ChannelUID(thing.getUID(), CHANNEL_COLORTEMPERATURE_ABS), stateDescriptionFragment);
-                }
-            }
-            capabilitiesInitializedSuccessfully = true;
+        if (capabilitiesInitializedSuccessfully || fullLight == null) {
+            return;
         }
+        Capabilities capabilities = fullLight.capabilities;
+        if (capabilities == null) {
+            return;
+        }
+        Control control = capabilities.control;
+        if (control == null) {
+            return;
+        }
+        ColorTemperature ct = control.ct;
+        if (ct == null) {
+            return;
+        }
+        colorTemperatureCapabilties = ct;
+
+        // minimum and maximum are inverted due to mired/Kelvin conversion!
+        StateDescriptionFragment stateDescriptionFragment = StateDescriptionFragmentBuilder.create()
+                .withMinimum(new BigDecimal(LightStateConverter.miredToKelvin(ct.max))) //
+                .withMaximum(new BigDecimal(LightStateConverter.miredToKelvin(ct.min))) //
+                .withStep(new BigDecimal(100)) //
+                .withPattern("%.0f K") //
+                .build();
+        stateDescriptionProvider.setStateDescriptionFragment(
+                new ChannelUID(thing.getUID(), CHANNEL_COLORTEMPERATURE_ABS), stateDescriptionFragment);
+
+        capabilitiesInitializedSuccessfully = true;
     }
 
     @Override

